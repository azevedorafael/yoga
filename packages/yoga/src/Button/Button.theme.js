import { hexToRgb } from '@gympass/yoga-common';

const Button = ({
  spacing,
  fontSizes,
  borders,
  radii,
  colors,
  fontWeights,
}) => ({
  padding: {
    right: spacing.xlarge,
    left: spacing.xlarge,
  },
  height: {
    default: 48,
    small: 32,
  },
  font: {
    size: fontSizes.small,
    weight: fontWeights.bold,
  },
  border: {
    small: {
      width: borders.small,
    },
    default: {
      width: borders.medium,
    },
    radius: radii.circle,
  },
  types: {
    contained: {
      backgroundColor: {
        disabled: colors.gray[3],
        default: colors.primary[3],
        pressed: colors.primary[2],
      },
      font: {
        default: {
          color: colors.white,
        },
        disabled: {
          color: colors.gray[7],
        },
        pressed: {
          color: colors.white,
        },
      },
    },
    outline: {
      backgroundColor: {
        disabled: 'transparent',
        default: 'transparent',
        pressed: 'transparent',
        hover: hexToRgb(colors.primary[3], 0.3),
      },
      font: {
        default: {
          color: colors.primary[3],
        },
        disabled: {
          color: colors.gray[7],
        },
        pressed: {
          color: colors.primary[2],
        },
      },
    },
    text: {
      backgroundColor: {
        disabled: 'transparent',
        default: 'transparent',
        pressed: 'transparent',
        hover: hexToRgb(colors.primary[3], 0.3),
      },
      font: {
        default: {
          color: colors.primary[3],
        },
        disabled: {
          color: colors.gray[7],
        },
        pressed: {
          color: colors.primary[2],
        },
      },
    },
    link: {
      font: {
        disabled: {
          color: colors.gray[7],
        },
      },
      margin: {
        top: spacing.medium,
        bottom: spacing.xsmall,
      },
    },
  },
});

export default Button;
