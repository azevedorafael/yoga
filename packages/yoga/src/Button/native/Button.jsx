import React from 'react';
import styled from 'styled-components';
import { node, bool } from 'prop-types';
import withTouchable from './withTouchable';

const Label = styled.Text`
  text-align: center;
  ${({
    disabled,
    pressed,
    inverted,
    theme: {
      yoga: {
        components: { button },
      },
    },
  }) => `
      font-size: ${button.font.size};
      font-weight: ${button.font.weight};
      color: ${button.types.contained.font.default.color};

      ${disabled ? `color: ${button.types.contained.font.disabled.color};` : ''}
      ${
        !disabled && pressed
          ? `color: ${button.types.contained.font.pressed.color};`
          : ''
      }

      ${
        inverted && !disabled
          ? `
        color: ${button.types.contained.backgroundColor.default};

        ${
          !disabled && pressed
            ? `color: ${button.types.contained.backgroundColor.pressed};`
            : ''
        }
      `
          : ''
      }
    `}
`;

const ButtonContainer = styled.View`
  ${({
    pressed,
    disabled,
    full,
    small,
    inverted,
    theme: {
      yoga: {
        components: { button },
      },
    },
  }) => `
      background-color: ${button.types.contained.backgroundColor.default};
      border-radius: ${button.border.radius}px;
      height: ${small ? button.height.small : button.height.default};
      justify-content: center;
      padding-left: ${button.padding.left}px;
      padding-right: ${button.padding.right}px;
      ${full ? 'width: 100%;' : ''}

      ${
        !disabled && pressed
          ? `
        background-color: ${button.types.contained.backgroundColor.pressed};
      `
          : ''
      }

      ${
        disabled
          ? `
        background-color: ${button.types.contained.backgroundColor.disabled};
      `
          : ''
      }

      ${
        inverted && !disabled
          ? `
          background-color: ${button.types.contained.font.default.color};
        `
          : ''
      }
    `}
`;

/** Buttons make common actions more obvious and help users more easily perform
them. Buttons use labels and sometimes icons to communicate the action that will
occur when the user touches them. */
const Button = ({
  children,
  full,
  disabled,
  small,
  pressed,
  inverted,
  ...rest
}) => (
  <ButtonContainer
    {...rest}
    full={full}
    pressed={pressed}
    disabled={disabled}
    small={small}
    inverted={inverted}
  >
    <Label disabled={disabled} pressed={pressed} inverted={inverted}>
      {children}
    </Label>
  </ButtonContainer>
);

Button.propTypes = {
  children: node,
  full: bool,
  disabled: bool,
  small: bool,
  pressed: bool,
  inverted: bool,
};

Button.defaultProps = {
  children: 'Button',
  full: false,
  disabled: false,
  small: false,
  pressed: false,
  inverted: false,
};

const ButtonWithTouchable = withTouchable(Button);

export { Label, ButtonContainer, ButtonWithTouchable as default };
