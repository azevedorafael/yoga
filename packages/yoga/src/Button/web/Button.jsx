import React from 'react';
import { node, func, bool } from 'prop-types';
import styled from 'styled-components';
import { hexToRgb } from '@gympass/yoga-common';

const StyledButton = styled.button`
  box-sizing: border-box;

  outline: none;
  transition: all 0.2s;
  cursor: pointer;

  ${({
    full,
    small,
    inverted,
    theme: {
      yoga: {
        components: {
          button: {
            padding: { left: paddingLeft, right: paddingRight },
            height: { small: smallHeight, normal: normalHeight },
            font: { size, weight },
            border: {
              small: { width: smallWidth },
              default: { width: defaultWidth },
              radius,
            },
            types: {
              contained: {
                backgroundColor: {
                  disabled: disabledBackgroundColor,
                  enabled: enabledBackgroundColor,
                  pressed: pressedBackgroundColor,
                },
                textColor: {
                  disabled: disabledTextColor,
                  enabled: enabledTextColor,
                  pressed: pressedTextColor,
                },
              },
            },
          },
        },
      },
    },
  }) => `
      width: ${full ? '100%' : 'auto'};
      height: ${small ? smallHeight : normalHeight}px;
      padding-left: ${paddingLeft}px;
      padding-right: ${paddingRight}px;

      background-color: ${enabledBackgroundColor};
      border: ${
        small ? smallWidth : defaultWidth
      }px solid ${enabledBackgroundColor};
      border-radius: ${radius}px;
      color: ${enabledTextColor};

      font-size: ${size}px;
      font-weight: ${weight};

      &:not([disabled]):hover, &:not([disabled]):focus {
        box-shadow: 0 4px 8px ${hexToRgb(enabledBackgroundColor, 0.45)};
      }

      &:active {
        background-color: ${pressedBackgroundColor};
        border-color: ${pressedBackgroundColor};
        color: ${pressedTextColor};
      }

      &:disabled {
        background-color ${disabledBackgroundColor};
        border-color: ${disabledBackgroundColor};
        color: ${disabledTextColor};

        cursor: not-allowed;
      }

      ${
        inverted
          ? `
        background-color: ${enabledTextColor};
        border-color: ${enabledTextColor};
        color: ${enabledBackgroundColor};

        &:active {
          background-color: ${enabledTextColor};
          border-color: ${enabledTextColor};
          color: ${pressedBackgroundColor};
        }

        &:not([disabled]):hover, &:not([disabled]):focus {
          box-shadow: 0 4px 8px ${hexToRgb(enabledTextColor, 0.45)};
        }
      `
          : ''
      }
    `}
`;

/** Buttons make common actions more obvious and help users more easily perform them. Buttons use labels and sometimes icons to communicate the action that will occur when the user touches them. */
const Button = ({
  children,
  onClick,
  full,
  disabled,
  inverted,
  small,
  ...props
}) => (
  <StyledButton
    disabled={disabled}
    full={full}
    inverted={inverted}
    onClick={onClick}
    small={small}
    {...props}
  >
    {children}
  </StyledButton>
);

Button.propTypes = {
  children: node,
  disabled: bool,
  full: bool,
  inverted: bool,
  onClick: func,
  small: bool,
};

Button.defaultProps = {
  children: 'Button',
  disabled: false,
  full: false,
  inverted: false,
  onClick: () => {},
  small: false,
};

export default Button;
