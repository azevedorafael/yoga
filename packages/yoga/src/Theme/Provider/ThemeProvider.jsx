import React from 'react';
import { ThemeProvider as SCThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import * as tokens from '@gympass/yoga-tokens';
import * as themes from '../themes';

const getTheme = ({ theme, locale }) => {
  const token = tokens[locale] || tokens.default;
  const appTheme = themes[theme] || themes.default;

  return appTheme(token);
};

const ThemeProvider = ({ children, ...theme }) => (
  <SCThemeProvider theme={{ yoga: getTheme(theme) }}>
    {children}
  </SCThemeProvider>
);

ThemeProvider.propTypes = {
  theme: PropTypes.string,
  locale: PropTypes.string,
  children: PropTypes.node.isRequired,
};

ThemeProvider.defaultProps = {
  theme: 'endUser',
  locale: 'pt-BR',
};

export default ThemeProvider;
