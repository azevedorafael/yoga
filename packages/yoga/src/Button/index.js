import Button from './web/Button';
import ButtonOutline from './web/Outline';
import ButtonText from './web/Text';
import ButtonLink from './web/Link';

Button.Outline = ButtonOutline;
Button.Text = ButtonText;
Button.Link = ButtonLink;

export default Button;
