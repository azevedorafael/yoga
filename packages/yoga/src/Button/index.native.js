import Button from './native/Button';
import ButtonOutline from './native/Outline';
import ButtonText from './native/Text';

Button.Outline = ButtonOutline;
Button.Text = ButtonText;

export default Button;
