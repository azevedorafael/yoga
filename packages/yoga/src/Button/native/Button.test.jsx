import React from 'react';
import { render, fireEvent, toJSON } from '@testing-library/react-native';
import ThemeProvider from '../../Theme';
import Button from '..';

describe('<Button />', () => {
  describe('Snapshots', () => {
    describe('Without props', () => {
      it('should match snapshot with default Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with outline Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Outline />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with text Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Text />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with link Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Link />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });
    });

    describe('With inverted prop', () => {
      it('should match snapshot with default Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button inverted />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with outline Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Outline inverted />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with text Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Text inverted />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with link Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Link inverted />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });
    });

    describe('With small prop', () => {
      it('should match snapshot with default Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button small />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with outline Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Outline small />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with text Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Text small />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });
    });

    describe('With full prop', () => {
      it('should match snapshot with default Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button full />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with outline Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Outline full />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with text Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Text full />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });
    });

    describe('With disabled prop', () => {
      it('should match snapshot with default Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button disabled />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with outline Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Outline disabled />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with text Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Text disabled />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });

      it('should match snapshot with link Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button.Link disabled />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });
    });
  });

  describe('With variant prop', () => {
    it('should match snapshot with default Button', () => {
      const { container } = render(
        <ThemeProvider>
          <Button.Link variant="primary" />
          <Button.Link variant="secondary" />
        </ThemeProvider>,
      );
      expect(toJSON(container)).toMatchSnapshot();
    });
  });

  describe('onPress prop', () => {
    it('should call onPress function when click on Button', () => {
      const onPressMock = jest.fn();
      const { getByText } = render(
        <ThemeProvider>
          <Button onPress={onPressMock}>Button</Button>
        </ThemeProvider>,
      );

      fireEvent.press(getByText('Button'));

      expect(onPressMock).toHaveBeenCalled();
    });

    it('should call onPress function when press on Button.Outline', () => {
      const onPressMock = jest.fn();
      const { getByText } = render(
        <ThemeProvider>
          <Button.Outline onPress={onPressMock}>Button.Outline</Button.Outline>
        </ThemeProvider>,
      );

      fireEvent.press(getByText('Button.Outline'));

      expect(onPressMock).toHaveBeenCalled();
    });

    it('should call onPress function when press on Button.Text', () => {
      const onPressMock = jest.fn();
      const { getByText } = render(
        <ThemeProvider>
          <Button.Text onPress={onPressMock}>Button.Text</Button.Text>
        </ThemeProvider>,
      );

      fireEvent.press(getByText('Button.Text'));

      expect(onPressMock).toHaveBeenCalled();
    });

    it('should not call onPress function when press on Button disabled', () => {
      const onPressMock = jest.fn();
      const { getByText } = render(
        <ThemeProvider>
          <Button disabled onPress={onPressMock}>
            Button
          </Button>
        </ThemeProvider>,
      );

      fireEvent.press(getByText('Button'));

      expect(onPressMock).not.toHaveBeenCalled();
    });
  });
});
