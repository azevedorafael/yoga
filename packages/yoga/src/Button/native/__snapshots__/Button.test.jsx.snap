// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshots With disabled prop should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    disabled={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        disabled={true}
        full={false}
        inverted={false}
        pressed={false}
        small={false}
        style={
          Array [
            Object {
              "backgroundColor": "#DBDBE8",
              "borderRadius": 9999,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          disabled={true}
          inverted={false}
          pressed={false}
          style={
            Array [
              Object {
                "color": "#9898A6",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        >
          Button
        </Text>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With disabled prop should match snapshot with link Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    disabled={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        disabled={true}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#9898A6",
              "fontSize": 14,
              "fontWeight": "700",
              "marginBottom": 8,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
        variant="primary"
      />
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With disabled prop should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    disabled={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        disabled={true}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#9898A6",
              "borderRadius": 9999,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          disabled={true}
          pressed={false}
          style={
            Array [
              Object {
                "color": "#9898A6",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With disabled prop should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    disabled={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        disabled={true}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 9999,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          disabled={true}
          pressed={false}
          style={
            Array [
              Object {
                "color": "#9898A6",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With full prop should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    full={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        disabled={false}
        full={true}
        inverted={false}
        pressed={false}
        small={false}
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 9999,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
              "width": "100%",
            },
          ]
        }
      >
        <Text
          disabled={false}
          inverted={false}
          pressed={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        >
          Button
        </Text>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With full prop should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    full={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        full={true}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#F46152",
              "borderRadius": 9999,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
              "width": "100%",
            },
          ]
        }
      >
        <Text
          pressed={false}
          style={
            Array [
              Object {
                "color": "#F46152",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With full prop should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    full={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        full={true}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 9999,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
              "width": "100%",
            },
          ]
        }
      >
        <Text
          pressed={false}
          style={
            Array [
              Object {
                "color": "#F46152",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With inverted prop should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    inverted={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        disabled={false}
        full={false}
        inverted={true}
        pressed={false}
        small={false}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 9999,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          disabled={false}
          inverted={true}
          pressed={false}
          style={
            Array [
              Object {
                "color": "#F46152",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        >
          Button
        </Text>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With inverted prop should match snapshot with link Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    inverted={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        disabled={false}
        inverted={true}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "fontWeight": "700",
              "marginBottom": 8,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
        variant="primary"
      />
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With inverted prop should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    inverted={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        inverted={true}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#FFFFFF",
              "borderRadius": 9999,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          inverted={true}
          pressed={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With inverted prop should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    inverted={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        inverted={true}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 9999,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          pressed={false}
          style={
            Array [
              Object {
                "color": "#F46152",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With small prop should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    small={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        disabled={false}
        full={false}
        inverted={false}
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 9999,
              "height": 32,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          disabled={false}
          inverted={false}
          pressed={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        >
          Button
        </Text>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With small prop should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    small={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#F46152",
              "borderRadius": 9999,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 32,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          pressed={false}
          style={
            Array [
              Object {
                "color": "#F46152",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With small prop should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    small={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 9999,
              "height": 32,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          pressed={false}
          style={
            Array [
              Object {
                "color": "#F46152",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots Without props should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        disabled={false}
        full={false}
        inverted={false}
        pressed={false}
        small={false}
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 9999,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          disabled={false}
          inverted={false}
          pressed={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        >
          Button
        </Text>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots Without props should match snapshot with link Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "marginBottom": 8,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
        variant="primary"
      />
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots Without props should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#F46152",
              "borderRadius": 9999,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          pressed={false}
          style={
            Array [
              Object {
                "color": "#F46152",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots Without props should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 9999,
              "height": 48,
              "justifyContent": "center",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          pressed={false}
          style={
            Array [
              Object {
                "color": "#F46152",
                "fontSize": 14,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> With variant prop should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    variant="primary"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "marginBottom": 8,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
        variant="primary"
      />
    </View>
  </TouchableWithoutFeedback>
  <TouchableWithoutFeedback
    variant="secondary"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#652CDB",
              "fontSize": 14,
              "fontWeight": "700",
              "marginBottom": 8,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
        variant="secondary"
      />
    </View>
  </TouchableWithoutFeedback>
  <TouchableWithoutFeedback
    variant="tertiary"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#2B9486",
              "fontSize": 14,
              "fontWeight": "700",
              "marginBottom": 8,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
        variant="tertiary"
      />
    </View>
  </TouchableWithoutFeedback>
</View>
`;
